src/Symfony/Component/AssetMapper/ImportMap/ImportMapUpdateChecker.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ImportMapUpdateChecker
+{
+    private const URL_PACKAGE_METADATA = 'https://registry.npmjs.org/%s';
+
+    public function __construct(
+        private readonly ImportMapConfigReader $importMapConfigReader,
+        private readonly HttpClientInterface $httpClient,
+    ) {
+    }
+
+    /**
+     * @param string[] $packages
+     *
+     * @return PackageUpdateInfo[]
+     */
+    public function getAvailableUpdates(array $packages = []): array
+    {
+        $entries = $this->importMapConfigReader->getEntries();
+        $updateInfos = [];
+        foreach ($entries as $entry) {
+            if (null === $entry->packageName || null === $entry->version) {
+                continue;
+            }
+            if (\count($packages) && !\in_array($entry->packageName, $packages, true)) {
+                continue;
+            }
+
+            $updateInfos[$entry->importName] = $this->getAvailableUpdateForEntry($entry);
```

Comment:
It makes the code a bit trickier, but we should try to make the requests in parallel - `JsDelivrEsmResolver` has an example of this. The flow would probably be:

A) Loop over `$entries`. Instead of an `$updateInfos` array set to the final project, we would probably have a `$responses` array set to the individual `Response` objects from making the HTTPClient request.

B) Then, do another `foreach` over `$responses` and process each one (get the latest version, determine the update type, etc).

This will allow all of the responses to start at once, and *then* we start processing them.

Reply:
Thanks for your time and advices @weaverryan !